hearts/hearts.py

```diff
@@ -215,6 +215,43 @@ def play_card(self, player, card):
         else:
             raise ValueError("Invalid play: it's not your turn")
 
+    def current_scores(self):
+        '''
+        Returns {player: int}.  Calculates the current number of points
+        for all tricks completed so far. Assumes the basic Hearts rules.
+        '''
+        scores = {player: 0 for player in self.players}
+        for trick in self.tricks:
+            scores[trick.winner()] += trick.points()
+        return scores
+
+    def shot_the_moon(self):
+        '''
+        Returns {player: Bool}.  Bool is True if and only if player's
+        score is 26 (based on the standard Hearts rules).  Function can
+        be called at any time during the round.
+        '''
+        d = {player: False for player in self.players}
+        scores = self.current_scores()
+        for player in self.players:
+            if scores[player] == 26:
+                d[player] = True
+                return d
+        return d
+
+    def final_scores(self):     # FIXME:  test_final_scores_shot_the_moon() fails. 
```

Comment:
Generally don't bother putting a FIXME because of a failing test. I won't let you merge a failing test anyway, and FXIME's tend to stick around even after they're fixed.